app/grandchallenge/reader_studies/tasks.py

```diff
@@ -102,12 +113,31 @@ def create_display_sets_for_upload_session(
 def add_image_to_answer(*, upload_session_pk, answer_pk):
     image = Image.objects.get(origin_id=upload_session_pk)
     answer = Answer.objects.get(pk=answer_pk)
+    question = answer.question
 
     if (
         str(answer.answer["upload_session_pk"]).casefold()
         == str(upload_session_pk).casefold()
     ):
-        add_image(answer, image)
+        image_file = image.files.filter(
+            image_type__in=[
+                ImageFile.IMAGE_TYPE_MHD,
+                ImageFile.IMAGE_TYPE_TIFF,
+            ]
+        ).get()
+        try:
+            question._validate_pixel_values(
```

Comment:
This is being done on the default queue. This task need to be re-worked so that it is idempotent and moved to the 2xlarge queue.